File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/DayFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class DayFunction : SingleArgumentFunction
    {
        public const string Name = "DAY";

        protected override string FunctionHelp => $"'{Name} expects 1 arguments (Date)";

        protected override VariableType ArgumentType => PrimitiveType.Date;
        protected override VariableType ResultType => PrimitiveType.Number;

        private DayFunction(Expression valueExpression, SourceReference reference)
            : base(valueExpression, reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference, Expression expression) =>
            new DayFunction(expression, reference);
    }
}

File: typescipt/src/language/expressions/functions/dayFunction.ts


export class DayFunction extends SingleArgumentFunction {
   public const string Name = `DAY`;

   protected override string FunctionHelp => $`'{Name} expects 1 argument (Date)`;

   protected override VariableType ArgumentType => PrimitiveType.Date;
   protected override VariableType ResultType => PrimitiveType.Number;

   constructor(valueExpression: Expression, reference: SourceReference)
     : base(valueExpression, reference) {
   }

   public static create(reference: SourceReference, expression: Expression): ExpressionFunction {
     return new DayFunction(expression, reference);
   }
}
